cps/kni.c

```diff
@@ -1123,6 +1124,94 @@ rd_getroute(const struct nlmsghdr *req, struct cps_config *cps_conf, int *err)
 	return MNL_CB_OK;
 }
 
+static void
+rd_fill_getlink_reply(struct cps_config *cps_conf,
+	struct mnl_nlmsg_batch *batch,
+	const char *kni_name, unsigned int kni_index, unsigned int kni_mtu,
+	uint32_t seq)
+{
+	struct nlmsghdr *reply;
+	struct ifinfomsg *ifim;
+
+	reply = mnl_nlmsg_put_header(mnl_nlmsg_batch_current(batch));
+	reply->nlmsg_type = RTM_NEWLINK;
+	reply->nlmsg_flags = NLM_F_MULTI;
+	reply->nlmsg_seq = seq;
+	reply->nlmsg_pid = cps_conf->nl_pid;
+
+	ifim = mnl_nlmsg_put_extra_header(reply, sizeof(*ifim));
+	ifim->ifi_family = AF_UNSPEC;
+	ifim->ifi_type = ARPHRD_ETHER;
+	ifim->ifi_index = kni_index;
+	ifim->ifi_flags = 0;
+	ifim->ifi_change = 0xFFFFFFFF;
+
+	mnl_attr_put_str(reply, IFLA_IFNAME, kni_name);
+	mnl_attr_put_u32(reply, IFLA_MTU, kni_mtu);
+}
+
+static int
+rd_getlink(const struct nlmsghdr *req, struct cps_config *cps_conf, int *err)
+{
+	char buf[2 * MNL_SOCKET_BUFFER_SIZE];
+	struct mnl_nlmsg_batch *batch;
+
+	batch = mnl_nlmsg_batch_start(buf, MNL_SOCKET_BUFFER_SIZE);
+	if (batch == NULL) {
+		CPS_LOG(ERR, "Failed to allocate a batch for a GETLINK reply\n");
+		*err = -ENOMEM;
+		goto out;
+	}
+
+	rd_fill_getlink_reply(cps_conf, batch,
+		rte_kni_get_name(cps_conf->front_kni),
+		cps_conf->front_kni_index,
+		cps_conf->net->front.vlan_insert
+			? cps_conf->net->front.mtu - sizeof(struct vlan_hdr)
+			: cps_conf->net->front.mtu,
```

Comment:
This `? :` expression should become an inline function defined in the previous patch `cps: configure MTU of KNI` and used here. Keep the comment explaining `- sizeof(struct vlan_hdr)` in the previous patch as well. This new function would receive `struct gatekeeper_if` and return the MTU of the corresponding KNI interface.